Azure provisioner Test panics in CI

When you run a test with the following property set in the properties file:

```toml
IS_CI_MANAGED_CLUSTER="true"
```

The code panics as below:

```bash
--- FAIL: TestCreateSimplePodAzure (55.14s)
    --- FAIL: TestCreateSimplePodAzure/SimplePeerPod_test (55.14s)
        --- FAIL: TestCreateSimplePodAzure/SimplePeerPod_test/PodVM_is_created (0.04s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x2e81e84]

goroutine 292 [running]:
testing.tRunner.func1.2({0x3198c80, 0x4ba0390})
	/opt/hostedtoolcache/go/1.20.6/x64/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.20.6/x64/src/testing/testing.go:1529 +0x39f
panic({0x3198c80, 0x4ba0390})
	/opt/hostedtoolcache/go/1.20.6/x64/src/runtime/panic.go:884 +0x213
github.com/confidential-containers/cloud-api-adaptor/test/e2e.AzureCloudAssert.HasPodVM({0xc0007fb770?}, 0xc0025e4ea0, {0xc000244513, 0x5})
	/home/runner/work/caa-test1/caa-test1/test/e2e/azure_test.go:68 +0x64
github.com/confidential-containers/cloud-api-adaptor/test/e2e.(*testCase).run.func2({0x3a36b50, 0xc000062158}, 0xc0025e4ea0, 0xc0000276c0?)
	/home/runner/work/caa-test1/caa-test1/test/e2e/common_suite_test.go:231 +0xa2b
sigs.k8s.io/e2e-framework/pkg/env.(*testEnv).executeSteps(0xc000246780, {0x3a36b50?, 0xc000062158?}, 0x4563dd8?, {0xc000765f50?, 0x1, 0x14dfc25?})
	/home/runner/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.1.0/pkg/env/env.go:421 +0xa3
sigs.k8s.io/e2e-framework/pkg/env.(*testEnv).execFeature.func1.1(0xc0025e4ea0)
	/home/runner/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.1.0/pkg/env/env.go:452 +0xad
testing.tRunner(0xc0025e4ea0, 0xc001884750)
	/opt/hostedtoolcache/go/1.20.6/x64/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.20.6/x64/src/testing/testing.go:1629 +0x3ea
FAIL	github.com/confidential-containers/cloud-api-adaptor/test/e2e	387.831s
FAIL
make: *** [Makefile:91: test-e2e] Error 1
Error: Process completed with exit code 2.
```

--- a/test/provisioner/provision_azure.go
+++ b/test/provisioner/provision_azure.go
@@ -34,6 +34,16 @@ func init() {
 func createResourceGroup() error {
 	if AzureProps.IsCIManaged {
 		log.Infof("Resource group %q is CI managed. No need to create new one manually.", AzureProps.ResourceGroupName)
+
+		resp, err := AzureProps.ResourceGroupClient.Get(context.Background(), AzureProps.ResourceGroupName, nil)
+		if err != nil {
+			err = fmt.Errorf("getting resource group %s: %w", AzureProps.ResourceGroupName, err)
+			log.Errorf("%v", err)
+			return err
+		}
+
+		AzureProps.ResourceGroup = &resp.ResourceGroup
+
 		return nil
 	}
 

